docs/articles/documentation/test-api/handling-native-dialogs.md

```diff
@@ -0,0 +1,141 @@
+---
+layout: docs
+title: Handling Native Dialogs
+permalink: /documentation/test-api/handling-native-dialogs.html
+---
+# Handling Native Dialogs
+
+TestCafe allows you to handle native browser dialogs whenever they are invoked during test run.
+You can close [alert](https://developer.mozilla.org/en-US/docs/Web/API/Window/alert) and
+[beforeunload](https://developer.mozilla.org/en-US/docs/Web/Events/beforeunload) dialogs,
+choose either option in [confirm](https://developer.mozilla.org/en-US/docs/Web/API/Window/confirm) dialogs
+or provide text to type into [prompt](https://developer.mozilla.org/en-US/docs/Web/API/Window/prompt) dialogs.
+
+You can also access the history of the invoked native dialogs so that you can assert that a certain dialog appeared or not.
+
+This topic contains the following sections:
+
+* [Dialog Handler](#dialog-handler)
+* [Dialog History](#dialog-history)
+
+## Dialog Handler
+
+To handle native dialogs invoked during the test run, specify a handler function
+by using the `setNativeDialogHandler` method of the
+[test controller](test-code-structure.md#test-controller).
+
+```text
+t.setNativeDialogHandler(fn(type, text), options)
+```
+
+Parameter  | Type                           | Description
+---------- | ------------------------------ | -------------
+`fn`       | Function &#124; ClientFunction | A regular or [client function](obtaining-data-from-the-client.md) that will be triggered whenever a native dialog is invoked.
+`options`  | Object                         | See [Client Function Options](obtaining-data-from-the-client.md#options).
+
+The handler function has the following arguments.
+
+Argument | Type   | Description
+-------- | ------ | -------------
+`type`   | String | The type of the native dialog. `'alert'` &#124; `'confirm'` &#124; `'beforeunload'` &#124; `'prompt'`.
+`text`   | String | Text of the dialog message.
+
+Once the handler is specified, it will be triggered each time a native dialog appears in the test.
+You can provide a new handler at any moment by calling `t.setNativeDialogHandler` once again.
+If a native dialog appears when no handler is set, the test fails with an error.
+
+To handle native dialogs that appear during the page load, specify the dialog handler
+before the first test action.
+
+> The handler is executed on the client side, so you cannot use Node.js API in the handler.
+
+You can control how a dialog is handled via the handler's return values.
+If you return nothing, TestCafe performs default handling.
+
+What kind of values should be returned depends on the dialog type. See the table below for reference.
```

Comment:
I think we should add a couple of words about native dialogs in iframes: we have one handler for all frames (top window's and iframes') and also one history stack for them all.